src/storage/src/source/reclock.rs

```diff
@@ -964,518 +685,457 @@ mod tests {
         // target upper (1001) because any new bindings will be minted beyond that timestamp.
-        let query = OffsetAntichain::from_iter([(PART_ID, MzOffset::from(4))]);
+        let query = partitioned_frontier([(PART_ID, MzOffset::from(4))]);
 
         assert_eq!(
             Ok(Antichain::from_elem(1001.into())),
-            follower.reclock_frontier(&query)
+            follower.reclock_frontier(query.borrow())
         );
     }
 
-    #[tokio::test(start_paused = true)]
-    async fn test_next_mint_timestamp() {
-        let now_fn = TestNowFn::new();
-        let persist_cache = persist_cache(now_fn.now_fn());
-
-        let (mut operator, mut follower) = make_test_operator(
-            ShardId::new(),
-            Antichain::from_elem(0.into()),
-            &persist_cache,
-            &now_fn,
-        )
-        .await;
-
-        // Test ceiling of timestamps works as expected
-        assert_eq!(operator.next_mint_timestamp(), Ok(0.into()));
-
-        now_fn.advance(Duration::from_millis(1));
-        assert_eq!(operator.next_mint_timestamp(), Ok(1000.into()));
-
-        now_fn.advance(Duration::from_millis(999));
-        assert_eq!(operator.next_mint_timestamp(), Ok(1000.into()));
-
-        now_fn.advance(Duration::from_millis(125));
-        assert_eq!(operator.next_mint_timestamp(), Ok(2000.into()));
-
-        // Advance the upper frontier to 2001
-        follower.push_trace_batch(operator.advance().await);
-
-        // Test calculation of sleep time works as expected
-        let sleep_duration = operator.next_mint_timestamp().unwrap_err();
-        assert_eq!(sleep_duration, Duration::from_millis(2001 - 1125));
-
-        // Test that if we wait the indicated amount we indeed manage to get a timestamp
-        now_fn.advance(sleep_duration);
-        assert_eq!(operator.next_mint_timestamp(), Ok(3000.into()));
-    }
-
-    #[tokio::test(start_paused = true)]
+    #[tokio::test]
     async fn test_reclock_frontier() {
-        let now_fn = TestNowFn::new();
-        let persist_cache = persist_cache(now_fn.now_fn());
-
         const PART1: PartitionId = PartitionId::Kafka(1);
         const PART2: PartitionId = PartitionId::Kafka(2);
 
-        let (mut operator, mut follower) = make_test_operator(
-            ShardId::new(),
-            Antichain::from_elem(0.into()),
-            &persist_cache,
-            &now_fn,
-        )
-        .await;
+        let (mut operator, mut follower) =
+            make_test_operator(ShardId::new(), Antichain::from_elem(0.into())).await;
 
-        let query = OffsetAntichain::new();
+        let query = Antichain::from_elem(Partitioned::minimum());
         // This is the initial source frontier so we should get the initial ts upper
         assert_eq!(
-            Ok(Antichain::from_elem(0.into())),
-            follower.reclock_frontier(&query)
+            Ok(Antichain::from_elem(1.into())),
```

Comment:
It's because now in `ReclockOperator::new` we always wait for one tick and initialize the shard by writing the minimum antichain at `IntoTime::minimum()`